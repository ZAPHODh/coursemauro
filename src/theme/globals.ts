'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html {
        font-family: ${montserrat.style.fontFamily};
        max-width: 100vw;
    }
    ,
    main {
        max-width: 100vw;
    }
    body {
        padding: 0;
        margin: 0;
        overflow: hidden;
    }
    ,
    input {
        font-family: inherit;
    }
    ,
    a {
        color: inherit;
    }
    ,
    * {
        box-sizing: border-box;
    }
    ,
    button {
        font-family: inherit;
    }
`}`
