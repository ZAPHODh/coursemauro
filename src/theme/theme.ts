import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
}
