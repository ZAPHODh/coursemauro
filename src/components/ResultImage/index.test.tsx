import { render } from '@testing-library/react'
import { ResultImage } from '.'
import { ThemeClient } from '../../theme/ThemeClient'
import mauroPic from '../../../public/images/mauro.webp'
describe('ResultImage', () => {
    it('should render the ResultImage', () => {
        const { getByRole } = render(
            <ThemeClient>
                <ResultImage img={mauroPic} alt="test" />
            </ThemeClient>
        )

        const heading = getByRole('img')

        expect(heading).toBeInTheDocument()
    })
    it('should render the alt properly', () => {
        const { getByRole } = render(
            <ThemeClient>
                <ResultImage img={mauroPic} alt="test" />
            </ThemeClient>
        )

        const img = getByRole('img')

        expect(img).toHaveAttribute('alt', 'test')
    })
    it('should take a resultImage snapshot', () => {
        const { getByRole } = render(
            <ThemeClient>
                <ResultImage img={mauroPic} alt="test" />
            </ThemeClient>
        )

        const heading = getByRole('img')

        expect(heading).toMatchSnapshot()
    })
})
