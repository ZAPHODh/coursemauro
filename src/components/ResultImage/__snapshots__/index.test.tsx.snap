// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultImage should take a resultImage snapshot 1`] = `
<img
  alt="test"
  class="sc-73d527aa-0 fHlBsz"
  data-nimg="1"
  decoding="async"
  height="40"
  loading="lazy"
  src="/img.jpg"
  srcset="/img.jpg 1x, /img.jpg 2x"
  style="color: transparent;"
  width="40"
/>
`;
